src/libwalletqt/Subaddress.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Subaddress.h"
+#include <QDebug>
+
+Subaddress::Subaddress(Monero::Subaddress *subaddressImpl, QObject *parent)
+  : QObject(parent), m_subaddressImpl(subaddressImpl)
+{
+    qDebug(__FUNCTION__);
+    getAll();
+}
+
+QList<Monero::SubaddressRow*> Subaddress::getAll(bool update) const
```

Comment:
I'm also not quite familiar with the GUI codebase, and I've designed these function interfaces by simply copying and adapting the existing practice. The same idiom

```c++
virtual std::vector<T*> getAll() const = 0;
```

is used repeatedly in wallet2_api.h, so starting to use refcount here (in order to avoid potential bugs) would imply that all other similar parts need to be updated as well. Since I haven't seen any crashes or issues with the current way, I'd rather keep it as is, and perhaps make a separate PR later that replaces all these pointers with refcount in order to be extra sure (but again, my familiarity with the GUI codebase is not enough for assessing if it actually is an improvement or not).